### PR TITLE
DAOS-6443 test: Expand cancel for ticket

### PR DESCRIPTION
**Pull request by mjean308**
Test-tag-hw-large: pr,hw,large soak_smoke

This ticket will add code to test.py to support
Cancel for ticket for specified parameters/conditions
in single variant, multiple parameter type testcases
like soak and iorsmall

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

File: src/tests/ftest/soak/soak_utils.py [hunk 1]
```diff
@@ -584,11 +586,21 @@ def create_ior_cmdline(self, job_spec, pool, ppn, nodesperjob):
...
                 for o_type in oclass_list:
-                    if api in ["HDF5-VOL", "HDF5", "POSIX"] and ppn > 16:
-                        ppn = 16
-                    # DAOS-6095
-                    if api in ["HDF5-VOL", "HDF5"] and t_size == "4k":
-                        t_size = "1m"
+                    # Cancel for ticket DAOS-6095
+                    cancel_api_6095 = ["HDF5-VOL", "HDF5", "POSIX"]
+                    if api in cancel_api_6095 and t_size == "4k" and (
+                            o_type in ["RP_2G1", 'RP_2GX']):
```
Comment (phender) [inline src/tests/ftest/soak/soak_utils.py:592]: Optional: PEP8 suggests something like this (https://www.python.org/dev/peps/pep-0008/#indentation):
```suggestion
                    if (api in ["HDF5-VOL", "HDF5", "POSIX"]
                            and t_size == "4k"
                            and o_type in ["RP_2G1", 'RP_2GX']):
```

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 2]
```diff
@@ -192,10 +194,35 @@ def check_variant_skip(self):
 
     # pylint: disable=invalid-name
     def cancelForTicket(self, ticket):
-        """Skip a test due to a ticket needing to be completed."""
-        return self.cancel("Skipping until {} is fixed.".format(ticket))
+        """Skip a test due to a ticket needing to be completed.
+
+        Args:
+            ticket (object): the ticket (str) or group of tickets (list)
```
Comment (phender) [inline src/tests/ftest/util/apricot/apricot/test.py:200]:
```suggestion
            ticket (object): the ticket (str) or group of tickets (set)
```

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 3]
```diff
@@ -192,10 +194,35 @@ def check_variant_skip(self):
...
+            ticket = ", ".join(ticket)
+        return self.cancel("Skipping until {} {} fixed.".format(ticket, verb))
     # pylint: enable=invalid-name
 
+    def add_cancel_ticket(self, ticket, reason=None):
+        """Skip a test due to a ticket needing to be completed.
+
+        Args:
+            ticket (object): the ticket (str) or group of tickets (list)
+                that cause this test case to be cancelled.
```
Comment (phender) [inline src/tests/ftest/util/apricot/apricot/test.py:218]:
```suggestion
            ticket (str): the ticket (str) used to cancel the test
```

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 4]
```diff
@@ -152,6 +152,8 @@ def __init__(self, *args, **kwargs):
         self.log.info("Job-ID: %s", self.job_id)
         self.log.info("Test PID: %s", os.getpid())
         self._timeout_reported = False
+        self._teardown_cancel = set()
```
Comment (phender) [inline src/tests/ftest/util/apricot/apricot/test.py:155]: I wonder if we should add a comment here:
```
        # When cancelling within a test variant, use 'add_cancel_ticket(<ticket>)' to add to this set.
```

**Review by phender — CHANGES_REQUESTED** [review]
Minor fixes if you need to make other changes.

**Review by shimizukko — DISMISSED** [review]
LGTM

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by phender — APPROVED** [review]

**Pull request merged**